src/CloudNative.CloudEvents/Http/HttpContentExtensions.cs

```diff
@@ -16,11 +16,15 @@ public static class HttpContentExtensions
         /// <summary>
         /// Converts a CloudEvent to <see cref="HttpContent"/>.
         /// </summary>
-        /// <param name="cloudEvent">CloudEvent</param>
+        /// <param name="cloudEvent">The CloudEvent to convert. Must not be null, and must be a valid CloudEvent.</param>
         /// <param name="contentMode">Content mode. Structured or binary.</param>
-        /// <param name="formatter">Event formatter</param>
+        /// <param name="formatter">The formatter to use within the conversion. Must not be null.</param>
         public static HttpContent ToHttpContent(this CloudEvent cloudEvent, ContentMode contentMode, CloudEventFormatter formatter)
         {
+            Preconditions.CheckNotNull(cloudEvent, nameof(cloudEvent));
```

Comment:
Should Http bindings be in a separate package like the other protocol bindings?

Reply:
The other ones all have dependencies outside the framework. (That's why the ASP.NET Core one is separate.) I think it makes sense to bundle it where that doesn't introduce any dependencies. Happy to discuss though - I don't think it's a slam-dunk.

Reply:
Yeah, I think it is a case of architectural purity vs convenience. If it is more convenient to leave in the main package, I am fine with it.